File: app/schemas/file.py
from datetime import datetime
from sqlmodel import SQLModel
from typing import Optional, List, Any
from pydantic import field_validator


class FileCreate(SQLModel):
    """ファイル作成スキーマ"""
    description: Optional[str] = None


class FileUpdate(SQLModel):
    """ファイル更新スキーマ"""
    description: Optional[str] = None


class FileRead(SQLModel):
    """ファイル読み取りスキーマ"""
    id: int
    user_id: int
    filename: str
    original_filename: str
    file_type: str
    file_size: int
    rows_count: Optional[int] = None
    columns_count: Optional[int] = None
    sheets_count: Optional[int] = None
    status: str
    description: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    last_accessed_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class FileDataResponse(SQLModel):
    """ファイルデータレスポンススキーマ"""
    headers: List[str]
    rows: List[List[Any]]
    total_rows: int


class FileListResponse(SQLModel):
    """ファイル一覧レスポンススキーマ"""
    files: List[FileRead]
    total: int
    page: int
    page_size: int

File: frontend/src/types/file.ts
export interface File {
  id: number
  user_id: number
  filename: string
  original_filename: string
  file_type: string
  file_size: number
  file_path: string
  rows_count: number | null
  columns_count: number | null
  sheets_count: number | null
  status: string
  description: string | null
  created_at: string
  updated_at: string | null
  last_accessed_at: string | null
}

export interface FileListResponse {
  items: File[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface FileDataResponse {
  headers: string[]
  rows: unknown[][]
  total_rows: number
}

export interface Template {
  id: number
  name: string
  description: string | null
  file_type: string
  file_path: string
  thumbnail_path: string | null
  is_public: boolean
  created_by: number | null
  created_at: string
}

export interface TemplateListResponse {
  items: Template[]
  total: number
}

export interface FileUpdateData {
  description?: string
}

export interface Statistics {
  total_files: number
  total_templates: number
}
